addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/XSLTInjection.java

```diff
@@ -0,0 +1,274 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2019 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.commons.httpclient.URIException;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.AbstractAppParamPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+
+/**
+ * Active scan rule which checks for signs of XSLT injection vulnerabilities with requests
+ *
+ * @author CaptainFreak
+ * @author ZainabAlShowely
+ */
+public class XSLTInjection extends AbstractAppParamPlugin {
+    private static final String MESSAGE_PREFIX = "ascanalpha.xsltinjection.";
+
+    private enum XSLTInjectionCheck {
```

Comment:
Maybe this would be better as `XSLTInjectionType`? Or something that more closely articulates what the enum members are?

Reply:
In fact maybe a custom enum could be created that includes the types, payloads, evidence, etc.

Similar to:
https://github.com/zaproxy/zap-extensions/blob/master/addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/PiiScanner.java#L49-L76

Reply:
That would be best, I attempted an inner class at first but it seems the class doesn't get picked up as a rule in that case. Will change!